### ISSUE TITLE
cleanlab.classification CleanLearning gives error for incompatible label array shape

### ISSUE THREAD

**Issue by fahadjahangir** [cleanlab/cleanlab#245]
I am trying to fit a TempCNN model (wrapped in scikeras.wrappers KerasClassifier) using CleanLearning.. 
TempCNN uses one hot encoding and  `y_train_one_hot` has a shape: `(14235, 10)`

`from scikeras.wrappers import KerasClassifier` 
`from cleanlab.classification import CleanLearning` 

`clf = KerasClassifier(Archi_TempCNN(X_train, nbclasses=y_train_one_hot.shape[-1]), epochs=10)`
`cl = CleanLearning(clf=clf, seed=SEED) `
`_ = cl.fit(X_train, y_train_serialized)`
 

CleaningLearning takes labels as an array of shape ``(N,)`` so if I pass `_ = cl.fit(X_train, y_train_one_hot)`  it gives the following error: 

![grafik](https://user-images.githubusercontent.com/15300417/165548880-3e29f25a-1db3-410c-a8d3-c697854d2f7f.png)

and if I pass the label as 1D array `(14235, )` it give me following error: 
![grafik](https://user-images.githubusercontent.com/15300417/165549306-e82aacd4-37a1-406a-80ef-05ee5548daa9.png)

What could be the possible solution to this?

**Comment by fahadjahangir** [comment]
yes my Keras model expects one-hot encoding. I don't think so I can modify anything in my `Archi_TempCNN`, here is the implementaion of model:

`import keras
from keras import layers
from keras import optimizers
from keras.regularizers import l2
from keras.layers import Input, Dense, Activation, BatchNormalization, Dropout, Flatten, Lambda, SpatialDropout1D, Concatenate
from keras.layers import Conv1D, Conv2D, AveragePooling1D, MaxPooling1D, GlobalMaxPooling1D, GlobalAveragePooling1D, GRU, Bidirectional
from keras.callbacks import Callback, ModelCheckpoint, History, EarlyStopping
from keras.models import Model, load_model
from keras.utils.np_utils import to_categorical
from keras import backend as K

#-----------------			DEEP LEARNING			--------------------
#-----------------------------------------------------------------------
#-----------------------------------------------------------------------

#-----------------------------------------------------------------------		
def conv_bn(X, **conv_params):	
    nbunits = conv_params["nbunits"];
    kernel_size = conv_params["kernel_size"];

    strides = conv_params.setdefault("strides", 1)
    padding = conv_params.setdefault("padding", "same")
    kernel_regularizer = conv_params.setdefault("kernel_regularizer", l2(1.e-6))
    kernel_initializer = conv_params.setdefault("kernel_initializer", "he_normal")

    Z = Conv1D(nbunits, kernel_size=kernel_size, 
            strides = strides, padding=padding,
            kernel_initializer=kernel_initializer,
            kernel_regularizer=kernel_regularizer)(X)

    return BatchNormalization(axis=-1)(Z) #-- CHANNEL_AXIS (-1)

#-----------------------------------------------------------------------		
def conv_bn_relu(X, **conv_params):
    Znorm = conv_bn(X, **conv_params)
    return Activation('relu')(Znorm)
    
#-----------------------------------------------------------------------		
def conv_bn_relu_drop(X, **conv_params):	
    dropout_rate = conv_params.setdefault("dropout_rate", 0.5)
    A = conv_bn_relu(X, **conv_params)
    return Dropout(dropout_rate)(A)

#-----------------------------------------------------------------------		
def fc_bn(X, **fc_params):
    nbunits = fc_params["nbunits"];
    
    kernel_regularizer = fc_params.setdefault("kernel_regularizer", l2(1.e-6))
    kernel_initializer = fc_params.setdefault("kernel_initializer", "he_normal")

    Z = Dense(nbunits, kernel_initializer=kernel_initializer, kernel_regularizer=kernel_regularizer)(X)
    return BatchNormalization(axis=-1)(Z) #-- CHANNEL_AXIS (-1)

#-----------------------------------------------------------------------		
def fc_bn_relu(X, **fc_params):	
    Znorm = fc_bn(X, **fc_params)
    return Activation('relu')(Znorm)

#-----------------------------------------------------------------------		
def fc_bn_relu_drop(X, **fc_params):
    dropout_rate = fc_params.setdefault("dropout_rate", 0.5)
    A = fc_bn_relu(X, **fc_params)
    return Dropout(dropout_rate)(A)

#-----------------------------------------------------------------------		
def softmax(X, nbclasses, **params):
    kernel_regularizer = params.setdefault("kernel_regularizer", l2(1.e-6))
    kernel_initializer = params.setdefault("kernel_initializer", "glorot_uniform")
    return Dense(nbclasses, activation='softmax', 
            kernel_initializer=kernel_initializer,
            kernel_regularizer=kernel_regularizer)(X)

#-----------------------------------------------------------------------	

def Archi_TempCNN(X, nbclasses):

    #-- get the input sizes
    m, L, depth = X.shape
    input_shape = (L,depth)

    #-- parameters of the architecture
    l2_rate = 1.e-6
    dropout_rate = 0.5
    nb_conv = 3
    nb_fc= 1
    nbunits_conv = 64 #-- will be double
    nbunits_fc = 256 #-- will be double

    # Define the input placeholder.
    X_input = Input(input_shape)

    #-- nb_conv CONV layers
    X = X_input
    for add in range(nb_conv):
        X = conv_bn_relu_drop(X, nbunits=nbunits_conv, kernel_size=5, kernel_regularizer=l2(l2_rate), dropout_rate=dropout_rate)
    #-- Flatten + 	1 FC layers
    X = Flatten()(X)
    for add in range(nb_fc):
        X = fc_bn_relu_drop(X, nbunits=nbunits_fc, kernel_regularizer=l2(l2_rate), dropout_rate=dropout_rate)

    #-- SOFTMAX layer
    out = softmax(X, nbclasses, kernel_regularizer=l2(l2_rate))

    # Create model.
    model = Model(inputs = X_input, outputs = out, name='Archi_3CONV64_1FC256')
    
    model.compile(optimizer=tf.keras.optimizers.Adam(0.0005), loss="categorical_crossentropy", metrics=["accuracy"])
    
    return model`

and to fit the model:
`model = Archi_TempCNN(X_train, nbclasses=y_train_one_hot.shape[-1])

history = model.fit(X_train, y_train_one_hot, epochs=10, batch_size=32,
                        validation_data=(X_test, y_test_one_hot),
                        verbose=1)`

so I am not sure how could I feed integer-coded labels to Keras model.

**Comment by anishathalye** [comment]
As I mentioned above, you can create a wrapper that converts from integer -> one-hot before feeding to the inner classifier. Here is an example. I have not tested it, because the code above isn't complete, missing some things like `y_train_serialized`, and I don't have the data. If you provide a complete code + data (or dummy data), I can test it locally.

```python
from sklearn.base import BaseEstimator

'''
Wraps a sklearn estimator that uses one-hot encoded data
and produces an sklearn estimator that uses integer-encoded data.
Translates integer-encoded data on the way in into one-hot encoded data.
'''
class OneHotToIntegerClassifier(BaseEstimator):
    def __init__(clf, nbclasses):
        self.clf = clf
        self.nbclasses = nbclasses

    def fit(clf, X, y, **kwargs):
        # convert integer to one-hot encoded labels for inner classifier
        y = to_categorical(y, self.nbclasses)
        self.clf.fit(X, y, **kwargs)

    def predict_proba(X, **kwargs):
        return self.clf.predict_proba(X, **kwargs)

    def predict(X, **kwargs):
        return self.clf.predict(X, **kwargs)

    def score(X, y, **kwargs):
        y = to_categorical(y, self.nbclasses)
        self.clf.score(X, y, **kwargs)

clf = OneHotToIntegerClassifier(
        KerasClassifier(Archi_TempCNN(X_train, nbclasses=y_train_one_hot.shape[-1]), epochs=10),
        nbclasses=y_train_one_hot.shape[-1],
      )
cl = CleanLearning(clf=clf, seed=SEED)
# assuming y_train_serialized is something like np.argmax(y_train_one_hot, axis=1)
_ = cl.fit(X_train, y_train_serialized)
```

**Comment by anishathalye** [comment]
Hoping the above code sample addresses your issue. If not, feel free to re-open!
